app/components/inputs/scalable-search-box.jsx

```
@@ -0,0 +1,32 @@
+ * @param {string} [props.placeholder] - Optional placeholder text for the input.
+ * @param {string} [props.text] - Optional initial text value for the input.
+ * @param {bool} [props.autoFocus] - Optional flag to set the input to autofocus.
+ * @param {React.MutableRefObject} [props.ref]
+ * @param {() => void} [onClick]
+ * @returns {JSX.Element} The rendered search box component.
+ */
+export default function ScalableSearchBox(props) {
+  const router = useRouter();
+  const pathname = usePathname();
+  return (
+    <div className={`grow self-stretch justify-stretch flex search-box rounded-full h-14 max-w-[720px] bg-[--md-sys-color-surface-container] text-[--md-sys-color-on-surface-container] ${props.className}`} role='search' onClick={()=>{if (pathname !== "/search") router.push("/search")}}>
```

Biome (1.9.4)

[error] Enforce to have the onClick mouse event with the onKeyUp, the onKeyDown, or the onKeyPress keyboard event.

Actions triggered using mouse events should have corresponding keyboard events to account for keyboard-only navigation.

(lint/a11y/useKeyWithClickEvents)